pkg/resource/deploy/deployment_executor.go

```diff
@@ -54,8 +57,10 @@ func (ex *deploymentExecutor) checkTargets(targets UrnTargets) error {
 
 	olds := ex.deployment.olds
 	var news map[resource.URN]bool
+	// Step generator may not have been initialized yet, in which case the news
+	// set will be empty anyway
 	if ex.stepGen != nil {
-		news = ex.stepGen.urns
+		news = ex.stepGen.GetURNs()
 	}
```

Comment:
if we wanted to encapsulate this in the step generator, we could just always call `GetURNs` and let the implementation handle the `nil` check

Reply:
Yeah, This is incorrect anyway, I have a follow on change that removes the `GetURNs` method and replaces it with a `HasURN` method which does the check correctly under the lock. I would rather add that as a following smaller PR than expand this one further

Reply:
I'm jsut going to fix this here